hack/lib/build/environment.sh

```diff
@@ -183,6 +184,12 @@ function os::build::environment::withsource() {
     tar -cf - "${excluded[@]}" . | docker cp - "${container}:${workingdir}"
   fi
 
+  local tmpdir="$(docker inspect -f '{{ range .Config.Env }}{{ if eq (index (split . "=")  0) "TMPDIR" }}{{ index (split . "=") 1 }}{{ end }}{{end}}' ${container})"
```

Comment:
Why does TMPDIR have to exist?  I'm really confused here.

The original reason TMPDIR in the image is /openshifttmp is because old versions of docker set a 64mb limit on it, which is far too small for tmp output.  If that's fixed in docker now, we can simply remove that setting and restore it to the default of /tmp.  

If there is still a limit, then we want to mount a reasonable TMPDIR in, in which case we don't care where on the FS is inside the container or what the container thinks the dir is.  We just want to create a volume and mount it to /tmp and set TMPDIR to /tmp.  No need to do this reading, or to make it fall within _output.